### PR TITLE
Added the ability to name saved queries for visualization

### PR DESCRIPTION
**Pull request by kennyatpillar**

### CODE REVIEW

File: src/pages/visualization-view/visualization-view.js [hunk 1]
```diff
@@ -39,7 +41,9 @@ const VisualizationView = (props) => {
   React.useEffect(() => { if (idFromUrl) load(idFromUrl) }, [idFromUrl])
   React.useEffect(() => { if (idFromState) history.push(linkUrl) }, [idFromState])
 
-  const handleSave = () => { setDialogOpen(true); save('Placeholder Title', query) }
+  const handleTitleChange = (event) => {console.log(event.target.value)}
```
Comment (ManApart) [inline src/pages/visualization-view/visualization-view.js]: Do we still want to log this?

File: src/pages/visualization-view/connector.js [hunk 2]
```diff
@@ -13,7 +13,8 @@ const mapStateToProps = state => {
         isSaving: visualizationSaving(state),
         isSaveFailure: visualizationSaveFailure(state),
         isSaveSuccess: visualizationSaveSuccess(state),
-        isSaveable: isVisualizationSaveable(state)
+        isSaveable: isVisualizationSaveable(state),
+        isTitleSet: visualizationTitle(state) != null
```
Comment (mcsearchin) [inline src/pages/visualization-view/connector.js]: This feels like overkill.

File: src/pages/visualization-view/visualization-view.js [hunk 3]
```diff
@@ -66,11 +70,20 @@ const VisualizationView = (props) => {
...
+                {isTitleSet ? 
+                <SaveIndicator saving={isSaving} success={isSaveSuccess} failure={isSaveFailure} linkUrl={linkUrl} /> 
+                :
+                <div>
+                  <b>Query Title: </b> 
+                  <input className="prompt" type="text" name="queryTitle" placeholder="Query Name" value={title} onChange={handleTitleChange} required="true"></input>
+                  <br/>
+                  <button className="save-button" onClick={handleSave} disabled={!isTitleSet}>Save</button>
+                  <button onClick={closeDialog}>Cancel</button>
+                </div>}
```
Comment (mcsearchin) [inline src/pages/visualization-view/visualization-view.js]: Seems like we should break this little title dialog out into its own component.

File: src/pages/visualization-view/connector.js [hunk 4]
```diff
@@ -13,14 +13,16 @@ const mapStateToProps = state => {
         isSaving: visualizationSaving(state),
         isSaveFailure: visualizationSaveFailure(state),
         isSaveSuccess: visualizationSaveSuccess(state),
-        isSaveable: isVisualizationSaveable(state)
+        isSaveable: isVisualizationSaveable(state),
+        updateQuery: visualizationID(state) !== undefined
```
Comment (LtChae) [inline src/pages/visualization-view/connector.js]: maybe rename to `shouldUpdateQuery` or `isQueryUpdateable` - Ben

Comment (LtChae) [reply src/pages/visualization-view/connector.js]: On top of that, combine in `updateQuery || props.isSaving` in here too - Ben and Tim

File: src/pages/visualization-view/visualization-view.js [hunk 5]
```diff
@@ -21,26 +21,46 @@ const VisualizationView = (props) => {
...
   React.useEffect(() => { if (idFromState) history.push(linkUrl) }, [idFromState])
 
-  const handleSave = () => { setDialogOpen(true); save('Placeholder Title', query) }
-  const closeDialog = () => { setDialogOpen(false) }
+
+  const handleTitleChange = (event) => {setQueryTitle(event.target.value)}
+  const handleUpdate = () =>  {
+    setDialogOpen(true)
+    if(shouldUpdateQuery) {
+      update(idFromState, title, query)
```
Comment (LtChae) [inline src/pages/visualization-view/visualization-view.js]: So, wait, we do the update here and if they click save? - Ben and Tim

File: src/pages/visualization-view/visualization-view.js (hunk 3 above)
Comment (dunson062786) [reply src/pages/visualization-view/visualization-view.js]: Talked to Tony and decided the juice was not worth the squeeze.

File: src/pages/visualization-view/connector.js (hunk 4 above)
Comment (dunson062786) [reply src/pages/visualization-view/connector.js]: Refactored to just not use updateQuery.

File: src/pages/visualization-view/visualization-view.js (hunk 5 above)
Comment (dunson062786) [reply src/pages/visualization-view/visualization-view.js]: we renamed it to make it clear and removed the update

File: src/pages/visualization-view/visualization-view.js [hunk 6]
```diff
@@ -21,26 +21,46 @@ const VisualizationView = (props) => {
...
 
   React.useEffect(() => { reset() }, [])
   React.useEffect(() => { if (idFromUrl) load(idFromUrl) }, [idFromUrl])
   React.useEffect(() => { if (idFromState) history.push(linkUrl) }, [idFromState])
 
-  const handleSave = () => { setDialogOpen(true); save('Placeholder Title', query) }
-  const closeDialog = () => { setDialogOpen(false) }
+
+  const handleTitleChange = (event) => {setQueryTitle(event.target.value)}
+  const handleUpdate = () =>  {
```
[the remark this answers is not in the document]
Comment (dunson062786) [reply src/pages/visualization-view/visualization-view.js]: changed

**Pull request merged**